stdlib/public/core/Array.swift

```diff
@@ -1228,6 +1253,14 @@ extension Array: RangeReplaceableCollection {
     defer {
       _endMutation()
     }
+      
+    let wasContiguous = newElements.withContiguousStorageIfAvailable {
```

Comment:
iirc redundant endMutation is harmless but I should still move this path up above the defer